frame/system/src/extensions/check_pov.rs

```diff
@@ -0,0 +1,136 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{Config, Pallet};
+use codec::{Decode, Encode};
+use frame_support::{
+	traits::Get,
+	weights::{DispatchInfo, PostDispatchInfo},
+};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{DispatchInfoOf, Dispatchable, SignedExtension},
+	transaction_validity::{InvalidTransaction, TransactionValidityError},
+};
+
+/// Proof-of-Validity resource limit check.
+#[derive(Encode, Decode, Clone, Eq, PartialEq, Default, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct CheckPov<T: Config + Send + Sync>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config + Send + Sync> CheckPov<T>
+where
+	T::Call: Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+{
+	/// Creates a new `SignedExtension` to check the PoV parameters of the extrinsic.
+	pub fn new() -> Self {
+		Self(Default::default())
+	}
+
+	/// Do the pre-dispatch checks.
+	///
+	/// It checks and notes the new PoV size.
+	pub fn do_pre_dispatch(
+		info: &DispatchInfoOf<T::Call>,
+	) -> sp_std::result::Result<(), TransactionValidityError> {
+		let next_size = Self::check_pov_size(info)?;
+
+		crate::PovSize::<T>::put(next_size);
+		Ok(())
+	}
+
+	fn check_pov_size(
+		info: &DispatchInfoOf<T::Call>,
+	) -> sp_std::result::Result<u32, TransactionValidityError> {
+		let size_limit = T::PovParams::get().max_size;
+		let current_size = Pallet::<T>::pov_size();
+		let added_size = info.pov_size;
```

Comment:
What is with the size of the extrinsic itself? It also contributes to the size of the PoV.